homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -192,9 +199,16 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up the Bayesian Binary sensor."""
+    """Set up the Bayesian Binary sensor from a yaml config."""
+    # In some cases async_setup_platform is called with an empty config
```

Comment:
Which cases would that be?

Reply:
So this seems to happen once on startup if there are any bayesian sensors configured, ConfigFlow or YAML. I do not know why.
Could this be related to our discussion about `async_setup()` above? Is there something in `default_config:` which is calling an empty `bayesian:` integration key? Something to do with Bayesian newly being a helper?

Reply:
I think this is also resolved in [08f277d](https://github.com/home-assistant/core/pull/122552/commits/08f277d5cd2625bae1cf321f1f23169a8fcdaf58)